src/ints/int10_modes.cpp

```diff
@@ -1760,7 +1902,172 @@ uint32_t VideoModeMemSize(uint16_t mode) {
 	return static_cast<uint32_t>(mem_bytes);
 }
 
-void INT10_SetupPalette() {
-	init_all_palettes();
+static cga_colors_t handle_cga_colors_prefs_tandy(const std::vector<std::string> cga_colors_prefs)
+{
+	constexpr auto default_brown_level = 0.5f;
+
+	auto brown_level = default_brown_level;
+
+	if (cga_colors_prefs.size() > 1) {
+		auto brown_level_pref = cga_colors_prefs[1];
+
+		if (float p; sscanf(brown_level_pref.c_str(), "%f", &p)) {
+			auto cga_colors = cga_colors_default;
+
+			brown_level = clamp(p / 100, 0.0f, 1.0f);
+
+			constexpr float max_green = 0x2a;
+			cga_colors[6].green = static_cast<uint8_t>(max_green *
+			                                           brown_level);
+			return cga_colors;
+		} else {
+			LOG_WARNING("INT10: Invalid brown level value for 'tandy' CGA colors: '%s', "
+			            "using default brown level of %3.2f",
+			            brown_level_pref.c_str(),
+			            default_brown_level * 100);
+		}
+	}
+	return cga_colors_ibm5153;
+}
+
+static cga_colors_t handle_cga_colors_prefs_ibm5153(const std::vector<std::string> cga_colors_prefs)
```

Comment:
For containers and compound types, pass in reference:

```c++
const std::vector<std::string>& cga_colors_prefs
```

Reply:
Keep forgetting :)

Reply:
np!